src/sds.h

```diff
@@ -95,6 +95,27 @@ static inline unsigned char sdsType(const_sds s) {
     return flags & SDS_TYPE_MASK;
 }
 
+/* Returns a user data bit stored in the SDS header by sdsSetAuxBit. The bit
+ * index is 0-4. Returns 0 or 1. Always returns 0 for SDS_TYPE_5. */
+static inline int sdsGetAuxBit(const_sds s, int bit) {
+    unsigned char flags = s[-1];
+    return sdsType(s) == SDS_TYPE_5 ? 0 : flags >> (SDS_TYPE_BITS + bit);
+}
+
+/* Stores a bit in an unused area in the SDS header, except for SDS_TYPE_5. The
+ * bit index is 0-4. The value is 0 or 1. The aux bits are lost if the SDS is
+ * auto-resized. */
```

Comment:
I agree this worth mentioning. the bits are more fit to be used for immutable strings (in that regard creating an "immutable string type" or "field" type would make sense, but maybe we can do this refactor in a separate PR.

Reply:
Yeah, this is already quite low level. I'll add a sentence like *This is only for special uses like immutable SDS embedded in other structures.*